### ISSUE TITLE
Generation: add repetition penalty, logit bias, and sampling extensions

### ISSUE THREAD

**Issue by anibjoshi** [strata-ai-labs/strata-inference#63]
## Context

The current sampling implementation supports temperature, top-k, and top-p. Several commonly-used sampling features are missing.

### 1. Repetition penalty
Penalize tokens that have already appeared in the generated sequence. Standard in llama.cpp (`--repeat-penalty`), HuggingFace (`repetition_penalty`), and most inference servers.

```rust
pub struct SamplingConfig {
    // ... existing fields ...
    pub repeat_penalty: f32,      // 1.0 = disabled, >1.0 = penalize repeats
    pub repeat_last_n: usize,     // Window size for repeat tracking (0 = full context)
}
```

### 2. Frequency and presence penalty
OpenAI-style penalties based on token frequency in the generated text.

```rust
    pub frequency_penalty: f32,   // 0.0 = disabled, penalizes by count
    pub presence_penalty: f32,    // 0.0 = disabled, penalizes by presence (binary)
```

### 3. Logit bias
Allow callers to boost or suppress specific tokens by adding bias to logits before sampling.

```rust
    pub logit_bias: HashMap<u32, f32>,  // token_id → bias to add
```

### 4. Min-P sampling
Newer sampling method that filters tokens below a fraction of the top token's probability.

```rust
    pub min_p: f32,  // 0.0 = disabled, filters tokens with p < min_p * p_max
```

## Priority

Repetition penalty is the most impactful — without it, models tend to get stuck in repetition loops during long generation. The others are nice-to-have for API compatibility.

## Reference

- llama.cpp sampling: `common/sampling.cpp`
- OpenAI API spec: frequency_penalty, presence_penalty, logit_bias
